src/main/java/Menu.java

```diff
@@ -1,40 +1,60 @@
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
+import java.util.*;
+import java.util.function.Supplier;
 
 public class Menu {
 
     private UserInterface gameUI;
-
-    public enum Options {
-        TWO_HUMANS, HUMAN_AND_ROBOT, TWO_ROBOTS, HUMAN_AND_ALIEN
-    }
+    private HashMap<String, Supplier<List<Player>>> menuOptions;
+    private ArrayList<String> printedMenuOptions;
 
     public Menu(UserInterface gameUI) {
-        this.gameUI = gameUI;
+        this.gameUI        = gameUI;
+        menuOptions        = new HashMap<>();
+        printedMenuOptions = new ArrayList<>();
+
+        initializeOptions();
     }
 
     public List<Player> createPlayers() {
+        String option = setOption();
+        return menuOptions.get(option).get();
+    }
+
+    private void initializeOptions() {
+        menuOptions.put("1", this::twoHumans);
+        printedMenuOptions.add(" 1) Human vs. human\n");
+
+        menuOptions.put("2", this::humanAndRobot);
+        printedMenuOptions.add(" 2) Human vs. robot\n");
+
+        menuOptions.put("3", this::twoRobots);
+        printedMenuOptions.add(" 3) Robot vs. robot\n");
+
+        menuOptions.put("4", this::humanAndAlien);
+        printedMenuOptions.add(" 4) Human vs. alien\n");
+    }
```

Comment:
They are used in `setOption()`, where I get the option from the user input, and in `isInvalidOption(String option)` where I check that the option is correct. Will try to find a better way to handle it :+1: